plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -172,10 +172,13 @@ public function getOptOutView()
 
         $setCookieInNewWindow = Common::getRequestVar('setCookieInNewWindow', false, 'int');
         if ($setCookieInNewWindow) {
+            $requestNonce = Common::getRequestVar('nonce', false);
+            $nonce = Nonce::isValidMd5($requestNonce) ? $requestNonce : '';
```

Comment:
can we maybe just call `verifyNonce`?

Reply:
`verifyNonce` checks more thinks and that would modify the business logic here. I'm not familiar as deeply with it, so I went with the safe solution. 
I haven't checked yet if we have tests or not for this part, but I can look into that and see if that passes that way or not.

Reply:
I would say it should work to use verifyNonce so it only works when the correct nonce is used etc. Might be worth a test since then it also checks directly that actually the correct nonce is used (the verify method won't discard it which is why it likely should work).

Reply:
Just thinking, if verifyNonce is used, would the optout form be a way to be able to guess the nonce? Since it's not removed after checking. And if so, and an attacker gets the nonce, can they do something w/ it?

Reply:
@diosmosis we can simply discard the nonce when it's not correct.

Reply:
I updated the pr using `verifyNonce` and discard it if it's not valid.